crates/jxl-render/src/blend.rs

```diff
@@ -333,6 +337,62 @@ pub fn patch(
     }
 }
 
+pub fn spline(
+    image_header: &Headers,
+    base_grid: &mut [SimpleGrid<f32>],
+    spline: Spline,
+) -> crate::Result<()> {
+    tracing::trace!("{}", spline);
+
+    let all_samples = spline.get_samples();
+    let arclength = all_samples.len() as f32 - 2.0 + all_samples.last().unwrap().lenght;
+    for (i, arc) in all_samples.iter().enumerate() {
+        let arclength_from_start = f32::min(1.0, (i as f32) / arclength);
+
+        let t = 31.0 * arclength_from_start;
+        let sigma = continuous_idct(&spline.sigma_dct, t);
+        let inv_sigma = 1.0 / sigma;
+        let values = [
+            continuous_idct(&spline.xyb_dct[0], t),
+            continuous_idct(&spline.xyb_dct[1], t),
+            continuous_idct(&spline.xyb_dct[2], t),
```

Comment:
Oops. All colour values should be multiplied by the intensity (`arc.length`).

Reply:
Oh I've missed that!